main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1191,6 +1194,34 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
     case WeededAst.Type.Or(tpe1, tpe2, loc) => freeVars(tpe1) ++ freeVars(tpe2)
   }
 
+  /**
+    * Returns the free vars and their inferred kinds in the given type `tpe0`, with `varKind` assigned if `tpe0` is a type var.
+    */
+  private def freeVarsWithKind(tpe0: WeededAst.Type, varKind: Kind = Kind.Star): List[(Name.Ident, Kind)] = tpe0 match {
```

Comment:
We were previously using default parameters in e.g. `Kind.scala`, but I stopped using them because I found it far better to be explicit about the kind. In other words, how about we write an inner recursive function that takes the argument explicitly and then the outer does not have to take the argument at all. If that makes sense? :)

Reply:
Yep, I agree.